File: src/Tree/BinaryTreePreorderTraversal/BinaryTreePreorderTraversal_Recursive.java
class Solution {
    List<Integer> res = new LinkedList<>();
    public List<Integer> preorderTraversal(TreeNode root) {
        dfs(root);
        return res;
    }
    private void dfs(TreeNode root){
        if(root == null) return;
        //First add current element
        res.add(root.val);
        //traverse left side
        dfs(root.left);
        //traverse right side
        dfs(root.right);
    }
}

File: src/Tree/BinaryTreePreorderTraversal/BinaryTreePreorderTraversal_Recursive.py
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        list: List[int] = []

        def dfs(node: Optional[TreeNode]):
            if node is None:
                return
            list.append(node.val)
            dfs(node.left)
            dfs(node.right)

        dfs(root)

        return list
